301-create-ad-forest-with-subdomain/azuredeploy.json

```diff
@@ -68,16 +67,14 @@
         "Standard_DS3_v2"
       ],
       "metadata": {
-        "description": "The size and type of the VM. *_DS* or *s types support Premium Disks."
+        "description": "The size and type of the VM. *_DS* or *s types support Premium Disks. Hint: the DS* series seem most reliable for this complex template."

```

Comment:
DS is no more or less reliable than any other size - what problem are we trying to solve here?

Reply:
Speed is the problem, the combined speed of CPU and disk. If it is too slow, the template times out on DSC stuff. Server 2016 is especially vulnerable for this.  Apparently it is very busy just after deployment, so busy the template gets too few resources. So, I tried to select sku and disk combinations that are fast enough, just to stabilize the template. This particular (fast) combination has not failed me yet. 

I never had any problem with 2012 R2 on slower disks and sku's. That worked on an A1 with SATA disks.

Reply:
that seems a bit odd, would imply that the DSC extension just doesn't work (which would be a regression because it used to).  For now, let's use standard_ssd, and set allowedValues to a set of vms that have the core/memory requirements for AD...  Update the description for the requirement, not the perception of reliability.  If we see regular failures we can escalate that...

Reply:
> that seems a bit odd, would imply that the DSC extension just doesn't work (which would be a regression because it used to).

It's not really the extension, it's the DSC code that is used to install and configure Active Directory. This is resource intensive. Windows Server 2016 simply seems a resource hog just after deployment.

I'll make the suggested change and test if Server 2016 is stable. Server 2019 should be OK.

Reply:
I tried three times with Windows 2016 on standard SSD. It succeeded twice, and failed once with a DSC timeout: 

![image](https://user-images.githubusercontent.com/15653857/70357827-28ed3d80-1878-11ea-8841-63b67ab054f3.png)

And that's typical. With full SSD, this does not happen -- at least I have not seen it yet. 

@bmoore-msft I'd like to move back to full SSD just to avoid disappointments. Thoughts?